### PR TITLE
pybind/mgr: mgr_module.py pep8/flake8 cleanups & tox updates

### PR DESCRIPTION
**Pull request by phlogistonjohn**
Update the *.py files in src/pybind/mgr to be pep8 clean (via flake8).

Update the tox.ini so these file will be checked by flake8 in the future.

Update the tox.ini file to have an improved default envlist. Use that envlist when running "make check" via cmake - much like #56304 did for the src/cephadm dir. This allows the python to be checked with a simple `tox r` that will match what the CI will run, hopefully leading to fewer surprises.

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests. There should be no functional changes in these patches.

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by phlogistonjohn** [comment]
I like to push PRs as draft so I can wait for the make check results before bothering people for reviews... and then promptly forget about the PR I waiting to see the results for until after the results have expired on jenkins.

**Comment by phlogistonjohn** [comment]
jenkins test api

File: src/pybind/mgr/CMakeLists.txt [hunk 1]
```diff
@@ -13,7 +13,7 @@ if(WITH_MGR_ROOK_CLIENT)
 endif()
 if(WITH_TESTS)
   include(AddCephTest)
-  add_tox_test(mgr ${CMAKE_CURRENT_SOURCE_DIR} TOX_ENVS py3 py37 mypy flake8 jinjalint nooptional)
+  add_tox_test(mgr ${CMAKE_CURRENT_SOURCE_DIR} TOX_ENVS __tox_defaults__)
```
Comment (adk3798) [inline src/pybind/mgr/CMakeLists.txt:16]: for my clarity, what does `__tox_defaults__` actually resolve to here?

Comment (phlogistonjohn) [reply src/pybind/mgr/CMakeLists.txt:16]: It's a special sentinel value that's passed to the tox runner script. If the argument is equal to that string it just executes whatever the tox.ini file has configured in the envlist... e.g. https://github.com/ceph/ceph/blob/2d03e9841fe077c52d316becade5a01b2eb2987e/src/pybind/mgr/tox.ini#L2

Comment (adk3798) [reply src/pybind/mgr/CMakeLists.txt:16]: okay, does that mean we're running check-black and check-isort now as part of make check with this change?

Comment (phlogistonjohn) [reply src/pybind/mgr/CMakeLists.txt:16]: Correct. That's an intended part of this set of patches: to get those running as part of our regular test suite. Note that these environments only do anything for the smb module at this time, if other mgr modules need to be covered by these we need to update the tox.ini.

**Review by adk3798 — APPROVED** [review]

**Comment by adk3798** [comment]
https://pulpito.ceph.com/adking-2024-06-28_04:58:18-orch:cephadm-wip-adk-testing-2024-06-27-0902-distro-default-smithi/

Failures all fell under

- cluster log failures (known issue, being cleaned up)
- mds_upgrade_Sequence (known issue)
- staggered upgrade with agent on (known issue)

Nothing to block merging

**Pull request merged**